### DISCUSSION TITLE
How to disable everything but syntax highlighting?

### DISCUSSION THREAD

**Discussion by nmz787** [antlr/intellij-plugin-v4#446] [Q&A]
It seems like the plugin is conflicting with my unit-test code that rebuilds the lexers/parses from G4 files, such that I've had to disable the plugin to keep from getting false parsing failures. But disabling the plugin causes the syntax highlighting to be lost.

How do I only enable syntax highlighting, without the other functionality of the plugin being enabled?

**Comment by bjansen** [comment]
Do you have a separate lexer grammar that is referenced with a `tokenVocab` option? If so, I think your problems might com from the plugin generating a `.tokens` file near your grammars. In that case, your build will also generate a similar file, which could cause strange errors I guess. 
If you could provide a bit more details about your setup (Maven/Gradle/other, are your grammars in a subpackage, what options are passed to ANTLR during the build etc), that could help me help you :)

**Reply by nmz787** [reply]
I don't know about `tokenVocab` but I am using two g4 files, one for lexer and one for parser. I don't think I pass any options to ANTLR during the build. Yes a `.tokens` file is generated in the same dir as the `g4` files, along with the `.interp` and `.py` files. No build system, I just have a Python file to call a command-line subprocess which calls java on ANTLR passing each g4 file, one at a time (lexer first, then parser).

**Comment by nmz787** [comment]
I see you closed this issue. It would be really nice to be able to disable the plugin features aside from syntax highlighting. The plugin seems to bug out a lot for us, and disabling it causes syntax highlighting to be unavailable

**Reply by bjansen** [reply]
The whole point of this plugin is to provide a bunch of features related to ANTLR4 grammars. If you need them but they don't work, then it's probably a bug and I would need more details on how to reproduce it. If you're not interested in these features, then I guess you don't need the plugin. Uninstall it and use something like a textmate bundle instead to higlight your grammars.

See https://github.com/textmate/antlr.tmbundle/blob/master/Syntaxes/ANTLR.plist and https://www.jetbrains.com/help/idea/textmate.html

**Reply by nmz787** [reply]
I /did/ provide as much info on the bugs I am facing, previously. Unfortunately I cannot release any testcases due to the nature of the work I was using ANTLR under, and their filesizes.

**Reply by nmz787** [reply]
And thanks for the textmate info, I've not heard of that tool/plugin before.
